llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -35,6 +35,9 @@ def int_dx_typedBufferLoad_checkbit
 def int_dx_typedBufferStore
     : DefaultAttrsIntrinsic<[], [llvm_any_ty, llvm_i32_ty, llvm_anyvector_ty]>;
 
+def int_dx_bufferUpdateCounter
+    : DefaultAttrsIntrinsic<[llvm_i32_ty], [llvm_any_ty, llvm_i32_ty], [IntrWriteMem]>;
```

Comment:
`IntrWriteMem` indicates that this only writes memory. Since it returns the original counter value, it does read some memory too.

Reply:
No attribute would indicate may read/write memory.  However, what about `IntrInaccessibleMemOrArgMemOnly`?  I wonder if that attribute would more accurately capture what we want.  This counter memory is unique and cannot alias any other memory operations in the module.